schema.prisma

```diff
@@ -0,0 +1,682 @@
+  lang       Language
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// 課程目標
+model SyllabusGoal {
+  /// 流水號
+  id             Int      @id
+  /// 大綱id
+  syllabusId     Int
+  /// 課程描述
+  describe       String
+  /// 課程要求
+  background     String
+  /// 課程目標
+  goal           String
+  /// 關鍵字
+  keyword        String
+  /// 指定閱讀
+  designatedRead String
+  /// 語系
+  lang           Language
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// 課程目標建立相關資料
+// 待確認我們是否會用到
+// 待確認我們是否會用到
+// 待確認我們是否會用到
+// model SyllabusGoalInfo {
+//   /// 流水號
+//   sn       Int    @id
+//   /// 學期，如103-1
+//   semester String @db.VarChar(6)
+//   /// 課程識別碼
+//   // CEIBA 課程網頁有非正式課程及正式課程兩種。非正式課程指不在正式課程中的課程。
+//   // 正式課程，
+//   // 課程識別碼如 101 80002
+
+//   // 非正式課程，
+//   // 課程識別碼，前三碼為XXX 80002
+//   courseNo   String @db.VarChar(10)
+//   /// 班級
+//   classNo    String @db.VarChar(5)
+//   /// 和課程進度表流水號相同値
+//   // 待確認用途
+//   syllabusSn Int
+//   /// CEIBA 專用之 6 碼課號
+//   // 開設 CEIBA 網頁之課程皆有唯一的 6 碼課號。
+//   // 未開設 CEIBA 課程網頁者，沒有。
+//   courseSn   String @db.VarChar(6)
+
+// }
+
+/// 評量方式
+model SyllabusRequest {
+  /// 流水號
+  id         Int      @id
+  /// 大綱id 
+  syllabusId Int
+  /// 評分方式項目
+  item       String   @db.VarChar(60)
+  /// 評量項目百分比
+  percent    Int
+  /// 說明
+  notes      String
+  /// 語系
+  lang       Language
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// office time
+model SyllabusOfficeTime {
+  /// 流水號
+  id         Int    @id
+  /// 大綱id 
+  syllabusId Int
+  /// 星期
+  // 例 : 3 為 星期三
+  week       String @db.VarChar(1)
+  /// 開始時間的"時"
+  startHour  String @db.VarChar(2)
+  /// 開始時間的"分"
+  startMin   String @db.VarChar(2)
+  /// 結束時間的"時" 
+  endHour    String @db.VarChar(2)
+  /// 結束時間的"分""
+  endMin     String @db.VarChar(2)
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// office time switch
+// 待確認作用
+model SyllabusOfficeTimeSwitch {
+  /// 流水號
+  id       Int    @id
+  /// 大綱id 
+  syllabusId Int @unique
+  /// 固定時間 或另約時間
+  fixedTime Boolean
+  comment   String
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// ceiba 教師帳號 / 人員編號轉換 
+// 查詢教師聯絡資料時使用
+// 從 tea_emp取得人員編號再來此檔取得帳號
+// 省略其他未使用欄位
+model SyllabusTeacerAccount {
+  /// 教師帳號
+  accout String @db.VarChar(16)
+  /// 人員編號
+  seq    String @db.VarChar(15)
+
+  @@unique([accout, seq])
+}
+
+/// 教師資訊
+model SyllabusTeacherInfo {
+  /// 帳號 
+  // todo : 待確認是否可以直接用seq欄位
+  userId    String @id @db.VarChar(50)
+  /// 分類，T:教職員
+  // 待確認是否有不是教師的類別
+  role      String @db.VarChar(1)
+  /// 中文名
+  // todo: 是否需要紀錄英文名
+  nameZhTw  String @db.VarChar(50)
+  /// 個人首頁
+  url       String @db.VarChar(255)
+  email     String @db.VarChar(255)
+  phone     String @db.VarChar(50)
+  cellPhone String @db.VarChar(50)
+  /// 辦公室  
+  address   String @db.VarChar(255)
+  /// 個人簡介
+  info      String
+  /// 檔名
+  imgfile   String @db.VarChar(64)
```

Comment:
Are we going to handle image uploads? If so, I am afraid that will be a very complex work under current infrastructure.

Also, If we have a self-hosted image API, we should refer to another stage. Ex. Save the permanent link of that file in the Image site. or something like id?

Reply:
Need to confirm with UI/UX team. The table comes from ceiba DB, where most teacher keeps it empty. I don't think it is worth the work under the current situation. Label as `future-work` maybe?

Reply:
Sounds good, `Future-work`. Please also open a ticket on Clickup.